src/Util/ClassSourceManipulator.php

```diff
@@ -178,6 +180,40 @@ public function addInterface(string $interfaceName)
         $this->updateSourceCodeFromNewStmts();
     }
 
+    public function addTrait($classTraitName)
+    {
+        $this->addUseStatementIfNecessary($classTraitName);
+
+        $classNode = $this->getClassNode();
+        $name = Str::getShortClassName($classTraitName);
+
+        if (empty($classNode->stmts)) {
+            $classNode->stmts[0] = new Node\Name('use '.$name.';');
```

Comment:
Do we need this `if` statement? If there are no `stmts` yet on the class, wouldn't the `array_unshift` on line 211 work like normal? This is not a spot where we need to optimize performance, so I don't mind to the work below, even if the stmts are empty.

Reply:
The problem is not about array_unshift or performance optimization but about the presence or not of nodes. If there is nodes, we need to add a "\n" for create blankline after our first use Trait but if stmts is empty. 

`  

        if ($traitNode) {
            foreach ($traitNode as $node) {
                if ($node->traits[0]->toString() === $name) {
                    return;
                }
            }
            array_unshift($classNode->stmts, new Node\Stmt\TraitUse([
                new Node\Name($name)
            ]));
        } else {
            array_unshift($classNode->stmts,
                new Node\Name('use '.$name.(empty($classNode->stmts) ? ";" : ";\n"))
            );
        } `